src/lib/test/DeriveTests.cpp

```diff
@@ -703,13 +704,31 @@ void DeriveTests::symDerive(CK_SESSION_HANDLE hSession, CK_OBJECT_HANDLE hKey, C
 			break;
 		case CKK_DES:
 			mechEncrypt.mechanism = CKM_DES_ECB;
+			#ifdef WITH_BOTAN
+			if (mechType == CKM_DES_CBC_ENCRYPT_DATA || mechType == CKM_DES3_CBC_ENCRYPT_DATA) {
+				CK_BYTE iv[8] = { 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08 };
+				mechEncrypt = { CKM_DES_CBC, iv, sizeof(iv) };
+			}
+			#endif
 			break;
 		case CKK_DES2:
 		case CKK_DES3:
 			mechEncrypt.mechanism = CKM_DES3_ECB;
+			#ifdef WITH_BOTAN
+			if (mechType == CKM_DES_CBC_ENCRYPT_DATA || mechType == CKM_DES3_CBC_ENCRYPT_DATA) {
+				CK_BYTE iv[8] = { 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08 };
+				mechEncrypt = { CKM_DES3_CBC, iv, sizeof(iv) };
+			}
+			#endif
 			break;
 		case CKK_AES:
 			mechEncrypt.mechanism = CKM_AES_ECB;
+			#ifdef WITH_BOTAN
+			if (mechType == CKM_AES_CBC_ENCRYPT_DATA) {
+				CK_BYTE iv[16] = { 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16 };
```

Comment:
I don't know what this is for, but if it's meant to be a continuous sequence from 0x1 to 0x16, then it's wrong. The numbers between 0xa and 0xf are missing.

Reply:
I am still investigating why this test does not work on the Encrypt/Decrypt step.

Reply:
Is the just arbitrary data for testing? So it's not as simple as this?

```suggestion
				CK_BYTE iv[16] = { 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f, 0x10 };
```

Reply:
Ok, I don't know. I just saw something that looked like a mistake. Ignore me.

Now, the other PR is merged.

Reply:
I don't know either. It is a very strange behavior but Encrypt/Decrypt tests work so that must be something I miss.